### DISCUSSION TITLE
Migrating Shares

### DISCUSSION THREAD

**Discussion by Animalmix55** [Web3Auth/Web3Auth#1301] [Help]
A question arose from Twitter dropping free support for their v1.1/2 APIs: what happens if we can no longer use a login provider? Right now, our platform uses custom JWTs to authenticate with Twitter (because we wanted to save money rather than using Auth0), but if we lose access to the Twitter API, how could we keep the lights on?

Does Web3Auth provide users a way to migrate the custom auth share to, say, a passwordless email login? Can I do something in my app to facilitate users migrating, or should I add a module to collect more shares from the user?

I expect that most of our users aren't privy enough to Web3 to understand entering the private key in MetaMask, so I wanted to check to see if there's a thought-out way to do something like this before we go live.

**Comment by YZhenY** [comment]
Hey there, there is a way to migrate keys, but it requires the user to login to both solutions to migrate their key and is quite custom. Disabling Twitter as a login ahead of time might be the simpler alternative in the case we don't want to pay
